platform/linux-generic/odp_packet_io.c

```diff
@@ -506,6 +535,28 @@ int odp_pktio_close(odp_pktio_t hdl)
 	return 0;
 }
 
+static int configure_tx_event_compl(pktio_entry_t *entry)
+{
+	odp_pool_param_t params;
+	const char *name_base = "_odp_pktio_tx_compl_pool_";
+	const size_t name_len = strlen(name_base) + strlen(XSTR(ODP_CONFIG_PKTIO_ENTRIES)) + 1;
+	char pool_name[name_len];
+
+	snprintf(pool_name, name_len, "%s%d", name_base, odp_pktio_index(entry->s.handle));
+	odp_pool_param_init(&params);
+
+	params.type = ODP_POOL_BUFFER;
+	params.buf.num = pktio_global->config.num_tx_compl_evs;
+	/* Room for odp_packet_hdr_t::user_ptr */
+	params.buf.size = sizeof(void *);
+	entry->s.tx_compl_pool = odp_pool_create(pool_name, &params);
```

Comment:
For better code readability, I'd add a tx_compl struct type containing void *user_ptr, and then use sizeof() that here (and cast buffer addr into that when accessing user_ptr elsewhere).

Packet interface could be reconfigured (... start() -> stop() -> config() -> start() ...), so here we would create the pool only if it's not yet created. Destroy can be in close().